test/e2e_tests/capp_revision_e2e_test.go

```diff
@@ -0,0 +1,73 @@
+package e2e_tests
+
+import (
+	"context"
+	"fmt"
+	mock "github.com/dana-team/container-app-operator/test/e2e_tests/mocks"
+	"github.com/dana-team/container-app-operator/test/e2e_tests/testconsts"
+	utilst "github.com/dana-team/container-app-operator/test/e2e_tests/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"strconv"
+)
+
+const (
+	MoreThanNumRevision = 12
+	NumOfRevisions      = 10
+)
+
+var _ = Describe("Validate CappRevision capp creation", func() {
+	It("Should validate CappRevison lifecycle based on Capp lifecycle", func() {
+		baseCapp := mock.CreateBaseCapp()
+		By("Creating Capp regular capp")
+		desiredCapp := utilst.CreateCapp(k8sClient, baseCapp)
+
+		Eventually(func() int {
+			cappRevisons, _ := utilst.GetCappRevisions(context.Background(), k8sClient, *desiredCapp)
+			return len(cappRevisons)
+		}, testconsts.Timeout, testconsts.Interval).ShouldNot(BeZero(), "Should create CappRevisions")
+
+		By("Updating Capp")
+		desiredCapp = utilst.GetCapp(k8sClient, desiredCapp.Name, desiredCapp.Namespace)
+		desiredCapp.Annotations = make(map[string]string)
+		desiredCapp.Annotations["test"] = "test"
+		utilst.UpdateCapp(k8sClient, desiredCapp)
+		Eventually(func() bool {
+			cappRevisons, _ := utilst.GetCappRevisions(context.Background(), k8sClient, *desiredCapp)
+			return len(cappRevisons) > 1
+		}, testconsts.Timeout, testconsts.Interval).Should(BeTrue(), "Should create new CappRevision")
+
+		By("Deleting Capp")
+		utilst.DeleteCapp(k8sClient, desiredCapp)
+		Eventually(func() int {
+			cappRevisons, _ := utilst.GetCappRevisions(context.Background(), k8sClient, *desiredCapp)
+			return len(cappRevisons)
+		}, testconsts.Timeout, testconsts.Interval).Should(BeZero(), "Should delete all CappRevisions")
+
+	})
+
+	It("Should limit CappRevisions to 10 per Capp", func() {
```

Comment:
Use `fmt.Sprintf` to change the number 10 from hard-coded to be based on a variable since this const might change

Reply:
In the future we'll need to move this '10' to a configMap so it can be controller from outside the code.